### PR TITLE
Import tests from dpstd library - part 4/16

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Part 4/16
Please see details in part 1/16 - https://github.com/oneapi-src/oneDPL/pull/885
This PR should be merged into main only after https://github.com/oneapi-src/oneDPL/pull/885

### CODE REVIEW

File: test/xpu_api/iterators/iterator.range/begin-end.pass.cpp [hunk 1]
```diff
@@ -31,26 +38,18 @@
...
-// Disable the missing braces warning for this reason.
-#include "disable_missing_braces_warning.h"
-
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/iterator.range/begin-end.pass.cpp:51]: It seems, `sycl_read ` is not used..

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/make_move_iterator.pass.cpp [hunk 2]
```diff
@@ -16,42 +23,37 @@
...
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#define TEST_IGNORE_NODISCARD (void)
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/make_move_iterator.pass.cpp:37]: 1. It seems, `sycl_read ` is not used..
2. Probably, it makes sense using `sycl::access::mode::write` directly in one place below, w/o additional constexpr variable.

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/minus.pass.cpp [hunk 3]
```diff
@@ -18,72 +25,64 @@
...
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/minus.pass.cpp:37]: 1. It seems, `sycl_read ` is not used..
2. Probably, it makes sense using `sycl::access::mode::write` directly in one place below, w/o additional constexpr variable.

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_eq.pass.cpp [hunk 4]
```diff
@@ -17,43 +24,36 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_eq.pass.cpp:36]: 1. It seems, `sycl_read ` is not used..
2. Probably, it makes sense using `sycl::access::mode::write` directly in one place below, w/o additional constexpr variable.

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/cstddef>
+#include <oneapi/dpl/functional>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp:43]: It seems, sycl_read and sycl_write  are not used..

**Review by MikeDvorskiy — CHANGES_REQUESTED** [review]

File: test/xpu_api/iterators/iterator.range/begin-end.pass.cpp (hunk 1 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/iterator.range/begin-end.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/make_move_iterator.pass.cpp (hunk 2 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.nonmember/make_move_iterator.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_eq.pass.cpp (hunk 4 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_eq.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.plus/difference_type.pass.cpp [hunk 6]
```diff
@@ -15,75 +15,67 @@
...
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.plus/difference_type.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.star/op_star.pass.cpp [hunk 7]
```diff
@@ -14,23 +21,16 @@
...
-#    include <iterator>
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.star/op_star.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp (hunk 5 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_gte.pass.cpp [hunk 8]
```diff
@@ -17,43 +24,36 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_gte.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_lt.pass.cpp [hunk 9]
```diff
@@ -17,42 +24,36 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_lt.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_lte.pass.cpp [hunk 10]
```diff
@@ -17,43 +24,36 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_lte.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp [hunk 13]
```diff
@@ -16,31 +23,24 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/default1.pass.cpp [hunk 14]
```diff
@@ -14,61 +21,55 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/default1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/post1.pass.cpp [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/post1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.incr/post.pass.cpp [hunk 17]
```diff
@@ -14,31 +14,24 @@
...
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.incr/post.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.incr/pre.pass.cpp [hunk 19]
```diff
@@ -14,31 +14,24 @@
...
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.incr/pre.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_neq.pass.cpp [hunk 21]
```diff
@@ -17,43 +24,36 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.comp/op_neq.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.equal/move_iterator.pass.cpp [hunk 22]
```diff
@@ -17,32 +24,25 @@
...
-#    include <type_traits>
-namespace s = std;
-#endif
+#include "support/test_iterators.h"
+#include "support/utils.h"
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.equal/move_iterator.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp [hunk 23]
```diff
@@ -0,0 +1,124 @@
+// -*- C++ -*-
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp:1]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp]: difference_type4.pass.cpp => difference_type_minus.pass.cpp

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp [hunk 25]
```diff
@@ -0,0 +1,89 @@
+// -*- C++ -*-
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp [hunk 26]
```diff
@@ -0,0 +1,83 @@
+//===----------------------------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp]: difference_type3.pass.cpp => difference_type_plus.pass.cpp

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/post1.pass.cpp [hunk 27]
```diff
@@ -0,0 +1,90 @@
+// -*- C++ -*-
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/post1.pass.cpp]: Fixed

[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/post1.pass.cpp]: post1.pass.cpp => post_decr.pass.cpp

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp (hunk 25 above)
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.decr/pre1.pass.cpp]: pre1.pass.cpp => pre_decr.pass.cpp

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp [hunk 28]
```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp]: difference_type2.pass.cpp => difference_type_idx.pass.cpp

[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp [hunk 29]
```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp (hunk 26 above)
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.plus/difference_type3.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp [hunk 30]
```diff
@@ -0,0 +1,78 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of iterator check #4 in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp]: => `Wrong result of move_iterator check with minus operator in kernel_test()`

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp (hunk 29 above)
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp]: difference_type1.pass.cpp => difference_type_minus.pass.cpp

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp [hunk 31]
```diff
@@ -0,0 +1,86 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "Wrong result of iterator check #3 in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp]: => "Wrong result of move_iterator check with operator[] in kernel_test()"

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/iter.pass.cpp [hunk 32]
```diff
@@ -59,26 +57,25 @@ kernel_test()
...
+main()
 {
+#if TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of iterator check #2 in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/iter.pass.cpp:77]: => "Wrong result of move_iterator check with value constructor in kernel_test()"

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp [hunk 33]
```diff
@@ -70,27 +68,26 @@ kernel_test()
...
+main()
 {
+#if TEST_DPCPP_BACKEND_PRESENT
     auto ret = kernel_test();
-    if (ret)
-        std::cout << "Pass" << std::endl;
-    else
-        std::cout << "Fail" << std::endl;
-    return 0;
+    EXPECT_TRUE(ret, "Wrong result of iterator check #1 in kernel_test()");
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp:89]: => "Wrong result of move_iterator check with copy constructor in kernel_test()"

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp (hunk 23 above)
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iterator/types2.pass.cpp:1]: types2.pass.cpp = > types_move_it.pass.cpp

**Review by MikeDvorskiy — CHANGES_REQUESTED** [review]

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp (hunk 33 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/convert.pass.cpp:89]: Fixed.

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/iter.pass.cpp (hunk 32 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.const/iter.pass.cpp:77]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp (hunk 24 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.compound.assign.minus/difference_type4.pass.cpp]: Renamed to another name.

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp (hunk 30 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.minus/difference_type1.pass.cpp]: Fixed

File: test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp (hunk 31 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type2.pass.cpp]: Fixed

**Pull request merged**